openpype/hosts/houdini/api/lib.py

```diff
@@ -649,3 +655,71 @@ def get_color_management_preferences():
         "display": hou.Color.ocio_defaultDisplay(),
         "view": hou.Color.ocio_defaultView()
     }
+
+
+def publisher_show_and_publish():
+    """Open publisher window and trigger publishing action."""
+
+    main_window = get_main_window()
+    publisher_window = get_tool_by_name(
+        tool_name="publisher",
+        parent=main_window
+    )
+
+    publisher_window.set_current_tab("publish")
+    publisher_window.make_sure_is_visible()
+    publisher_window._reset_on_show = False
+
+    publisher_window._controller.reset()
+    publisher_window._controller.publish()
+
+
+def self_publish():
+    """Self publish from ROP nodes."""
+
+    current_node = hou.node(".").path()
+
+    host = registered_host()
+    context = CreateContext(host, reset=True)
+
+    for instance in context.instances:
+        node_path = instance.data.get("instance_node")
+        if not node_path:
+            continue
+
+        active = current_node == node_path
+        if not active:
+            continue
+
+        instance["active"] = active
+
+        result, comment = hou.ui.readInput(
+            "Add Publish Note",
+            buttons=("Ok", "Cancel"),
+            title="Publish Note",
+            close_choice=1
+        )
+        if not result:
+            instance.data["comment"] = comment
```

Comment:
With this you will set the comment on instance, but the comment value will stay there forever and user won't have any option to remove it.

Reply:
are there any alternatives ?

Reply:
Don't use per-instance comment.

Reply:
how can I set this comment from my python code? 
![image](https://github.com/ynput/OpenPype/assets/20871534/d10256ab-d2b0-4bd7-922f-f128cd04a0fc)

Reply:
I found that input text. I will push commit shortly.